cl_sii/rut/crypto_utils.py

```diff
@@ -0,0 +1,47 @@
+import re
+
+import cryptography
+import cryptography.x509
+from cryptography.hazmat.backends.openssl import backend as crypto_x509_backend
+
+from . import constants
+from . import Rut
+
+
+def get_subject_rut_from_certificate_pfx(pfx_file_bytes: bytes, password: str) -> Rut:
+    """
+    Return the Chilean RUT stored in a digital certificate.
+
+    :param pfx_file_bytes: Digital certificate in PKCS12 format
+    :param password: (Optional) The password to use to decrypt the PKCS12 file
```

Comment:
What do you mean by *optional*? Do you mean that `password` can be `None`? Or that the `password` argument must not necessarily be provided?

Reply:
Good point, it means that we don't need to provide the password (`None`), it is only needed in cases where the PFX file is encrypted.
So I will make some adjustments to make the parameter `Optional[bytes]` instead of `str`. I had kept it of type `str` to maintain compatibility with what we use in `FD-CL-Data`, but I think it would be no problem to make this small change.
https://github.com/fyntex/fd-cl-data/blob/a784817adc4d6f98e1de42396d832a93fff0fccd/fd_cl_data/apps/sii_auth/models/sii_auth_credential.py#L701

Ref: https://cryptography.io/en/latest/hazmat/primitives/asymmetric/serialization/#cryptography.hazmat.primitives.serialization.pkcs12.load_key_and_certificates
> password (bytes-like) – The password to use to decrypt the data. `None` if the PKCS12 is not encrypted.

Reply:
I've been thinking about the developer experience aspects of using `bytes` for the password, and I think that the argument would be developer-friendlier if its type is `str` (or actually `Optional[str]`), and the conversion to bytes is performed inside the method (`password.encode()`).

Reply:
> I've been thinking about the developer experience aspects of using `bytes` for the password, and I think that the argument would be developer-friendlier if its type is `str` (or actually `Optional[str]`), and the conversion to bytes is performed inside the method (`password=password.encode() if password is not None else None,`)

The above comment is also applicable to `SiiAuthSession.log_in_with_certificate(..., passphrase)` from https://github.com/fyntex/lib-cl-sii-api-python/pull/288.